cli/cli.go

```diff
@@ -68,7 +73,48 @@ type ServerConfig struct {
 	// AdminSecret (optional) Admin secret required to query the endpoint
 	AdminSecret string `yaml:"admin_secret,omitempty"`
 
-	ParsedEndpoint *url.URL `yaml:"-"`
+	ParsedEndpoint             *url.URL                   `yaml:"-"`
+	HasuraServerInternalConfig HasuraServerInternalConfig `yaml:"-"`
+}
+
+func (c *ServerConfig) GetHasuraInternalServerConfig() error {
+	// Determine from where assets should be served
+	url := c.Endpoint + "/v1alpha1/config"
+	client := http.Client{Timeout: 30 * time.Second}
+	req, err := http.NewRequestWithContext(context.Background(), "GET", url, nil)
+	if c.AdminSecret != "" {
+		req.Header.Set("x-hasura-admin-secret", c.AdminSecret)
+	}
+	if err != nil {
```

Comment:
shouldn't the error check happen before setting `adminSecret` header?

Reply:
🤦‍♂ Yes!